File: SearchInsertPosition.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l = 0
        r = len(nums) - 1
        if target <= nums[l]:
            return l
        if target == nums[r]:
            return r
        if target > nums[r]:
            return r+1
        while True:
            mid = (l + r) //2
            if nums[mid] ==target:
                return mid
            elif nums[mid] >target:
                r = mid
            else:
                l = mid
            if r-l <= 1:
                return l + 1

File: SearchInsertPosition.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1, m;
        if (target <= nums[l])
            return l;
        else if (target > nums[r])
            return r + 1;
        else if (target == nums[r])
            return r;
        while (1){
            m = (l + r) / 2;
            if(nums[m] == target)
                return m;
            else if (nums[m] < target)
                l = m;
            else
                r = m;
            if (r-l == 1)
                return l+1;
        }
    }
};